### ISSUE TITLE
FTL strings from shared components are duplicated in Pontoon

### ISSUE THREAD

**Issue by vpomerleau** [mozilla/fxa#14548]
## Description

Fxa has recently started to share FTL strings between multiple components. The shared strings originate in `fxa-react` and `fxa-shared` (`branding.ftl`). Currently, the shared strings are contacted by `grunttask` into each package, then sent for localization. Concatenating the shared strings before localization has resulted in the same strings needing to be translated multiple times.

Discussion about this issue started in this thread: https://github.com/mozilla/fxa/pull/14439#issuecomment-1326807819

### Steps to reproduce

Example:
`app-footer-terms-of-service` from `fxa-react/components/Footer/en.ftl` is concatenated into `settings.ftl` and `payments.ftl`.
This string would show up twice for localization in Pontoon, once for each file.

### Expected result

- FTL strings that are shared by multiple components only need to be translated once by localizers.
- Where multiple components use a shared string, the localization is the same in all components.

### Actual result

FTL strings (from `fxa-shared` and `fxa-react`) that are shared by multiple packages are concated into each package's FTL file (e.g., `auth.ftl`, `settings.ftl`, `payments.ftl`) before being sent for localization, so translation need is duplicated/triplicated. 

Consequences:
- Workload for localizers is unnecessarily increased
- Risk that strings that are intended to be standardized across packages are actually translated differently for each package

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-6388)

**Comment by vpomerleau** [comment]
Below are my thoughts on a possible approach. Documenting here to check if this makes sense for both fxa and l10n ( @LZoog, @flodolo, @bcolsson, please review. I'm still new so I fully expect adjustments will be needed. Thanks!

fxa:

1. Remove concatenation of `branding.ftl` and `fxa-react` ftl strings from `auth.ftl`,`settings.ftl`, `payments.ftl`.
2. New gruntfile to concatenate `fxa-react` strings into a new `shared.ftl` file in `fxa-react`.

fxa-content-server-l10n:

3. Add steps in `/scripts/extract_strings.sh` to extract fluent files from `fxa-react` and `fxa-shared` to `templates`.

pontoon:

4. `Pontoon`: `branding.ftl` and `shared.ftl` are added to files to be localized for Firefox Accounts. - shared strings are translated only once.

fxa:

5.  Update `clone-l10n.sh` to copy `shared.ftl` and `branding.ftl` as needed into packages.

ex.

```
case "$MODULE" in
    "$PAYMENTS")
        copy_ftl "payments"
        copy_ftl "shared"
        copy_ftl "branding"
        ;; 
```

6. Add `shared`, `branding` to Fluent bundles for `fxa-auth-server`,`fxa-payments`, `fxa-settings`.

**Comment by flodolo** [comment]
Makes sense to me. 

We probably want to do something as a 3b step on our side, to recover the translations for `branding.ftl` from one of the existing files.

**Comment by bcolsson** [comment]
Agreed with @flodolo that this approach if it works on the FxA side would work for us.

I can create a script to generate the localized versions of `branding.ftl`.

**Comment by LZoog** [comment]
From #14439, @flodolo:

> Technically, it's not important how the files are used on the FxA side, what would be important for us is to have separate files for localization (branding, app, shared).
> 
> If these files are merged into one at build level, to avoid extra network requests, that would still work.

Interesting, I wasn't aware having a separate `branding.ftl` file would be preferable - it sounds like references to terms that don't exist in that file are OK and won't make a linter/validator complain? Or if it does, we're OK with that since we know the branding will exist in the end? If so, this all sounds good to me, as in the end it'll be less work on your side. 🙂

Merging the files together after our clone step seems fine to me. I'm currently unsure if we'll need to do this now, or if we can wait until we work on [FXA-2156](https://mozilla-hub.atlassian.net/browse/FXA-2156) (#9675). @vpomerleau if we can change the value of [`AppLocalizationProvider` bundles](https://github.com/mozilla/fxa/blob/74eeb4f3afd9a37d9d2b3c7bf2177cf77f74b8d2/packages/fxa-settings/src/index.tsx#L64) to `['branding', 'settings', 'shared']` after creating those files and it works without an extra script, let's do that for now and we can leave a note on FXA-2156 about optimizing that there.

> fxa:
> 
> 1. Remove concatenation of `branding.ftl` and `fxa-react` ftl strings from `auth.ftl`,`settings.ftl`, `payments.ftl`.
> 2. New gruntfile to concatenate `fxa-react` strings into a new `shared.ftl` file in `fxa-react`.

> 6. Add `shared`, `branding` to Fluent bundles for `fxa-auth-server`,`fxa-payments`, `fxa-settings`.

Just a couple of nitpicks: for 1 & 6, let's let SubPlat know about these plans (even just in a comment in a related ticket filed for them) since they can't reference that shared branding file yet. For 2., let's also modify the `## Brands used in fxa-auth-server.` comment in `branding.ftl`. Also for 6, we don't need `shared` in `fxa-auth-server`.

But, yeah. This sounds good to me, thanks for putting this together @vpomerleau. 👍

**Comment by vpomerleau** [comment]
Thanks @LZoog! 

For step 1, though we have a concatenation grunt task for the fxa-react strings, those haven't been merged on the l10n side yet. (see https://github.com/mozilla/fxa-content-server-l10n/pull/665)

For steps 5-6, to be clear, my plan is to bundle these files for now:
auth: `branding.ftl`, `auth.ftl` (doesn't need shared)
settings: `settings.ftl`, `shared.ftl` (branding.ftl will be included in a later step, with [PR-14442](https://github.com/mozilla/fxa/pull/14442))
payments: `payments.ftl`, `shared.ftl` (branding.ftl` to be added later, will require more cleanup)

I'll get started on a draft PR for all this so we can get those new strings merged in as soon as possible.

**Comment by flodolo** [comment]
> From #14439, @flodolo:
> 
> > Technically, it's not important how the files are used on the FxA side, what would be important for us is to have separate files for localization (branding, app, shared).
> > If these files are merged into one at build level, to avoid extra network requests, that would still work.
> 
> Interesting, I wasn't aware having a separate `branding.ftl` file would be preferable - it sounds like references to terms that don't exist in that file are OK and won't make a linter/validator complain? Or if it does, we're OK with that since we know the branding will exist in the end? If so, this all sounds good to me, as in the end it'll be less work on your side. 🙂

I don't know exactly how FxA loads Fluent files, but you normally can have multiple files in the bundle, and references are resolved at run-time.

FxA is the only project with Fluent that concatenate separate FTL files. See for example Relay
https://github.com/mozilla-l10n/fx-private-relay-l10n/tree/main/en

**Comment by vpomerleau** [comment]
Just put up draft PRs:
`fxa`: https://github.com/mozilla/fxa/pull/14567
`fxa-content-server-l10n`: https://github.com/mozilla/fxa-content-server-l10n/pull/667

> FxA is the only project with Fluent that concatenate separate FTL files. 

@flodolo We use separate `en.ftl` for each React component - concatenating on our end before sending to Pontoon ensures that the end result on Pontoon is similar to Relay's (instead of dozens of en.ftl files).

**Comment by flodolo** [comment]
This should be fixed at this point?
